pkg/upgrade/upgrades/tenant_table_migration_test.go

```diff
@@ -60,10 +61,21 @@ func TestUpdateTenantsTable(t *testing.T) {
 	var (
 		validationSchemas = []upgrades.Schema{
 			{Name: "name", ValidationFn: upgrades.HasColumn},
+			{Name: "data_state", ValidationFn: upgrades.HasColumn},
+			{Name: "service_mode", ValidationFn: upgrades.HasColumn},
 			{Name: "tenants_name_idx", ValidationFn: upgrades.HasIndex},
+			{Name: "tenants_service_mode_idx", ValidationFn: upgrades.HasIndex},
 		}
 	)
 
+	// Clear the initial KV pairs set up for the system tenant entry.
```

Comment:
If you take another iteration of the code can you put a TODO here that says we should be bootstrapping into the old version and avoid this dance altogether?